src/core/InputField/InputField.tsx

```diff
@@ -0,0 +1,174 @@
+        className={
+          disabled
+            ? classes.disabled
+            : `${classes.root}  ${getVariant(variant)}`
+        }
+      >
+        <InputLabel htmlFor="outlined-adornment-password">{label}</InputLabel>
+        <OutlinedInput
+          role="OutlinedInput"
+          type={showPassword ? 'text' : 'password'}
+          value={value}
+          error={variant === 'error' ? true : false}
+          disabled={disabled}
+          onChange={onChange}
+          required={required}
+          endAdornment={
+            <InputAdornment position="end">
+              <IconButton
+                aria-label="toggle password visibility"
+                onClick={handleClickShowPassword}
+                onMouseDown={handleMouseDownPassword}
+                edge="end"
+              >
+                {showPassword ? <Visibility /> : <VisibilityOff />}
+              </IconButton>
+            </InputAdornment>
+          }
+          startAdornment={
+            startIcon ? (
+              <InputAdornment position="start">
+                <IconButton aria-label="password field icon" edge="start">
+                  {startIcon}
+                </IconButton>
+              </InputAdornment>
+            ) : (
+              <></>
+            )
+          }
+          labelWidth={70}
+        />
+      </FormControl>
+    );
+  }
+  if (type === 'text') {
+    return (
+      <FormControl
+        data-testid="inputField"
+        variant="outlined"
+        className={
+          disabled
+            ? classes.disabled
+            : `${classes.root}  ${getVariant(variant)}`
+        }
+      >
+        <InputLabel htmlFor="outlined-adornment">{label}</InputLabel>
+        <OutlinedInput
+          role="OutlinedInput"
+          type={type}
+          value={value}
+          error={variant === 'error' ? true : false}
+          disabled={disabled}
+          onChange={onChange}
+          required={required}
+          startAdornment={
+            startIcon ? (
+              <InputAdornment position="start">
+                <IconButton edge="start">{startIcon}</IconButton>
+              </InputAdornment>
+            ) : (
+              <></>
+            )
+          }
+          endAdornment={
+            endIcon ? (
+              <InputAdornment position="end">
+                <IconButton edge="end">{endIcon}</IconButton>
+              </InputAdornment>
+            ) : (
+              <></>
+            )
+          }
+          labelWidth={70}
+        />
+      </FormControl>
+    );
+  } else {
+    return (
+      <FormControl
+        data-testid="inputField"
+        variant="outlined"
+        className={
+          disabled
+            ? classes.disabled
+            : `${classes.root}  ${getVariant(variant)}`
+        }
+      >
+        <InputLabel
+          htmlFor="outlined-adornment"
+          className={disabled ? `MuiInputLabel-shrink MuiFormLabel-filled` : ``}
```

Comment:
This last else is not needed....the type text already handles it
and this disable class should also be there for type text and password